benchmark/benchmark.js

```diff
@@ -0,0 +1,453 @@
+// @ts-check
+
+const fs = require('fs');
+const path = require('path');
+const crypto = require('crypto');
+const { argv } = require('yargs');
+const fetch = require('node-fetch').default;
+const Benchmark = require('benchmark');
+const simpleGit = require('simple-git/promise');
+const { parseLanguageNames } = require('../tests/helper/test-case');
+
+
+/**
+ * @param {import("./config").Config} config
+ */
+async function runBenchmark(config) {
+	const cases = await getCases(config);
+	const candidates = await getCandidates(config);
+	const maxCandidateNameLength = candidates.reduce((a, c) => Math.max(a, c.name.length), 0);
+
+	const totalNumberOfCaseFiles = cases.reduce((a, c) => a + c.files.length, 0);
+	console.log(`Found ${cases.length} cases with ${totalNumberOfCaseFiles} files in total.`);
+	console.log(`Test ${candidates.length} candidates with Prism.${config.options.testFunction}`);
+	const estimate = candidates.length * totalNumberOfCaseFiles * config.options.maxTime;
+	console.log(`Estimated duration: ${Math.floor(estimate / 60)}m ${Math.floor(estimate % 60)}s`);
+
+	/**
+	 * @type {Summary[]}
+	 *
+	 * @typedef {{ best: number; worst: number, relative: number[], avgRelative?: number }} Summary
+	 */
+	const totalSummary = Array.from({ length: candidates.length }).map(() => ({ best: 0, worst: 0, relative: [] }));
```

Comment:
Nit, but the second param to `Array.from` is a mapper function, so you don't need to call `.map` separately.

Reply:
Done.